auth/src/main/java/software/amazon/awssdk/auth/signer/internal/Aws4SignerRequestParams.java

```diff
@@ -64,51 +65,20 @@
      */
     private final String formattedSigningDate;
 
-    /**
-     * The signing algorithm to be used for computing the signature.
-     */
-    private final String signingAlgorithm;
-
-    /**
-     * The original modeled request given to the SDK.
-     */
-    private final SdkRequest originalRequest;
-
-    /**
-     * The HTTP request to be signed.
-     */
-    private final SdkHttpFullRequest.Builder httpRequest;
-
     /**
      * Generates an instance of AWS4signerRequestParams that holds the parameters used for computing a AWS 4 signature
      * for a request.
      */
-    @ReviewBeforeRelease("This should be simplified with the signer refactor.")
-    public Aws4SignerRequestParams(SdkRequest originalRequest, SdkHttpFullRequest.Builder httpRequest,
-                                   ExecutionAttributes executionAttributes,
-                                   Date signingDateOverride, String regionNameOverride,
-                                   String serviceName, String signingAlgorithm) {
-        if (signingAlgorithm == null) {
-            throw new IllegalArgumentException("Signing Algorithm cannot be null");
-        }
-        this.originalRequest = originalRequest;
+    public Aws4SignerRequestParams(SdkHttpFullRequest.Builder httpRequest, AwsSignerParams signerParams) {
```

Comment:
Passing around a mutable request is scary, to me, since it could be modified by multiple threads.

Reply:
We don't need the SdkHttpFullRequest or Builder here. Removed.